### DISCUSSION TITLE
How to Transfer amount to let the account doesn't remain dust balance

### DISCUSSION THREAD

**Discussion by BomaxChen** [filecoin-project/community#213] [Developers]
If I have filecoin account, its balance is 1 FIL, how do I use RPC API to get gas information that I can transfer all this account balance to another account ? (It means after this transaction, the previous account remains 0 FIL , that is I can calculate the real gas fee. )

I don't want to remain dust balance like ( 0.000000XXX FIL), so I need to transfer all amount from the previous account. Thanks a lot

**Comment by jimpick** [comment]
I know the Glif wallet does a nice gas calculation when you use it ... there's probably some clever code there. https://wallet.glif.io/

**Reply by jimpick** [reply]
Here's some of the code: https://github.com/glifio/wallet/blob/primary/components/Wallet/Send.js/CustomizeFee.jsx#L116

**Reply by jimpick** [reply]
Chasing the code further:

https://github.com/glifio/modules/blob/69f792fbb30980681e106b88201e1148eff29a9d/packages/filecoin-wallet-provider/src/filecoin.ts#L203

If you read through that, you can see it calls the `GasEstimateMessageGas` API method eventually (line 182).

Here's the automated docs for that ... you'll probably need to read some of the Lotus code if you really want to understand how it works:

https://docs.filecoin.io/reference/lotus-api/#gasestimatemessagegas

**Reply by jimpick** [reply]
The base fee fluctuates, so it's probably impossible to transfer every possible attoFIL out unless you are lucky.
